.github/workflows/pr_master_test.yml

```diff
@@ -456,16 +462,102 @@ jobs:
         make -C doc/OnlineDocs doctest -d
 
     - name: Process code coverage report
-      env:
-        CODECOV_NAME: ${{matrix.TARGET}}/${{matrix.python}}${{matrix.other}}
       run: |
         coverage combine
         coverage report -i
         coverage xml -i
+
+    - name: Record build artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: ${{github.job}}_${{env.GHA_JOBGROUP}}-${{env.GHA_JOBNAME}}
+        path: |
+          .coverage
+          coverage.xml
+          # In general, do not record test results as artifacts to
+          #   manage total artifact storage
+          # TEST-*.xml
+
+  cover:
+    name: process-coverage-${{ matrix.TARGET }}
+    needs: build
+    if: always() # run even if a build job fails
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+        include:
+        - os: ubuntu-latest
+          TARGET: linux
+        - os: macos-latest
+          TARGET: osx
+        - os: windows-latest
+          TARGET: win
+
+    steps:
+    - name: Checkout Pyomo source
+      uses: actions/checkout@v2
+      # We need the source for .codecov.yml and running "coverage xml"
+
+    - name: Pip package cache
+      uses: actions/cache@v1
+      id: pip-cache
+      with:
+        path: cache/pip
+        key: pip-${{env.CACHE_VER}}.0-${{runner.os}}-3.8
+
+    - name: Download build artifacts
+      uses: actions/download-artifact@v2
+      with:
+        path: artifacts
+
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - name: Install Python Packages (pip)
+      shell: bash # DO NOT REMOVE: see note above
+      run: |
+        python -c 'import sys;print(sys.executable)'
+        python -m pip install --cache-dir cache/pip --upgrade pip
+        pip install --cache-dir cache/pip ${PYTHON_CORE_PKGS} \
+            ${PYTHON_REQUIRED_PKGS}
+        python -c 'import sys; print("PYTHON_EXE=%s" \
+            % (sys.executable,))' >> $GITHUB_ENV
+
+    - name: Install Pyomo and PyUtilib
+      run: |
+        echo ""
+        echo "Clone Pyomo-model-libraries..."
+        git clone https://github.com/Pyomo/pyomo-model-libraries.git
+        echo ""
+        echo "Install PyUtilib..."
+        echo ""
+        $PYTHON_EXE -m pip install git+https://github.com/PyUtilib/pyutilib
+        echo ""
+        echo "Install Pyomo..."
+        echo ""
+        $PYTHON_EXE setup.py develop ${{matrix.setup_options}}
+        echo ""
+        echo "Set custom PYOMO_CONFIG_DIR"
+        echo ""
+        echo "PYOMO_CONFIG_DIR=${GITHUB_WORKSPACE}/config" >> $GITHUB_ENV
+
+    - name: Generate parse_table_datacmds
```

Comment:
I just don't know the purpose of this. What does having `parse_table_datacmds.py` do for us?

Reply:
`coverage xml` parses the source code to identify covered/uncovered lines and will die if the source is not found. `parse_table_datacmds` is an auto generated file, so we have to do something to make sure it exists.